applications/StructuralMechanicsApplication/custom_elements/base_solid_element.h

```diff
@@ -50,6 +50,68 @@ namespace Kratos
 class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION)  BaseSolidElement
     : public Element
 {
+protected:
+    /**
+     * Internal variables used in the kinematic calculations
+     */
+    struct KinematicVariables
+    {
+        Vector  StrainVector;
+        Vector  StressVector;
+        Vector  N;
+        Matrix  B;
+        double  detF;
+        Matrix  F;
+        double  detJ;
+        double  detJ0;
+        Matrix  J0;
+        Matrix  InvJ0;
+        Matrix  DN_DX;
+        Matrix  D;
+        
+        /**
+         * The variables are initialized
+         * @param StrainSize: The size of the strain vector in Voigt notation
+         * @param Dimension: The size of the strain vector in Voigt notation
+         */
+        void Initialize( 
```

Comment:
I was thinking of doing that, but I had problems of compilation, and it was already late (I commited this at mighnight)